### ISSUE TITLE
Subtitles missing for some programs

### ISSUE THREAD

**Issue by andreas1974** [kodi-svtplay/xbmc-svtplay#291]
# Problem description

When I watch some programs there are no subtitles.
# Steps to reproduce

* Search for and start "Avlyssningen" (the 1974 movie).
* Make sure subtitles are switched on.
* Subtitles are missing

# System information
Kodi version: 18.9
Platform: LibreElec 9.2.8
Plugin version: 5.1.21

# Log

**Comment by linqcan** [comment]
Sorry for the inconvenience and thanks for reporting. I will look into this in due time.

I saw you are using version 5.1.20, any reason you are not on 5.1.21?

**Comment by andreas1974** [comment]
Oh... I copied the facts from my previous post, and didn't recall there was an update. I checked the version now and it is indeed 5.1.21.  Sorry!
I changed it in my post now.

**Comment by gandalth** [comment]
Hi everyone, I also stumbled across this. I use the plugin to watch the live stream, e.g. SVT Barn, and I do not get any subtitles at all. Is this meant to work?

Here is my setup:
Raspberry Pi4
Libreelec 12 Nightly
Kodi 21 Beta2
Current Plugin Version from Kodi Repo

**Comment by linqcan** [comment]
It's meant to work I just haven't had time to get around looking at it.
There are some other changes needed as well that's in my plans.

**Comment by gandalth** [comment]
Perfect, I am happy to be your tester when you have time to commit an update. Feel free to ping me here.

**Comment by gandalth** [comment]
Oops, I had the wrong kodi version in my earlier comment. It is 21 Beta2. I fixed that in the original comment.

**Comment by linqcan** [comment]
So I have done som digging and concluded it's time to move to a new video player for this addon that can better support SVT Play's new default format DASH. It is a quite major overhaul and might affect backward compatability.

Unortunately there is no easy fix to get subtitles working for neither case reported here as the way subtitles are currently distributed is no supported out-of-the-box by Kodi.

Thanks again for reporting this and pushing me to look into this again!

Reference: https://github.com/xbmc/inputstream.adaptive/wiki/Integration

**Comment by gandalth** [comment]
Thank you for looking into this, much appreciated!

I experimented in parallel downloading a file from svtplay with yt-dlp. When I select subtitles in vtt format and provide them to kodi, playback is fine (with subtitles). Does that only work with files, but not with streaming segments?

**Comment by linqcan** [comment]
Just wanted to let you know that the re-work is happening (https://github.com/kodi-svtplay/xbmc-svtplay/tree/feature/inputstream.adaptive) but is going slowly due to life.

**Comment by gandalth** [comment]
Much appreciated!
Jag ska lärar mig svenska mellan, but I will still need the subtitles :)

**Comment by linqcan** [comment]
I think I have a good solution for Kodi 19+, but there are still some regrression testing to do since the plugin supports Kodi >=17 (krypton) and I don't want to break that right now.

If someone want to test it right now you can uninstall this plugin in Kodi and manually install the zip in [this](https://github.com/kodi-svtplay/xbmc-svtplay/releases/tag/ia-beta-1) release.

**Comment by gandalth** [comment]
I am happy to test and report back.

I just tried to install from zip file, and kodi tells me installation failed due to an invalid structure within the zip file.
I am on Kodi 21.0-Beta2.

**Comment by linqcan** [comment]
Thanks for testing out! That's odd, I will verify the zip. Don't hold for
me so please do reinstall the official version while I debug the issue.

**Comment by gandalth** [comment]
Thank you for looking into this. No rush!

**Comment by gandalth** [comment]
BTW, I ran the install once more and checked the log file on installing.
It sounds to me as if kodi wants to see a folder, which is missing:

Error <general>: CAddonInstaller: installing addon failed '/storage/downloads/kodi-svtplay-ia-beta1.zip' - itemsize: 7, first 
item is folder: false

**Comment by gandalth** [comment]
I was able to fix the installation issue by re-zipping the content (in one folder).

The plugin works, but
- for live programs, e.g. SVT Barn, playback always stops after around 30 seconds (maybe that is one dash segment?).
- Neither for live programming nor for recorded shows, subtitles work yet. I switch them on in the general plugin settings and in the stream settings. But the player does show 'SWE' for subtitle properties.

**Comment by linqcan** [comment]
Thanks for reporting this. I have tested on the stable Kodi release 20.x,
and had no issues. You tested on the beta version 21, right?

**Comment by gandalth** [comment]
Correct. I am using Beta2 of Kodi 21, and my system is one of libreelec's nightly builds (built Dec 16, 2023).

Let me do this: I will upgrade to libreelec's most recent nightly and retest. Beta2 of Kodi 21 is still current, but libreelec has pulled more recent, non-tagged versions from Kodi's repository since then.
I can only do that next week though.

**Comment by gandalth** [comment]
I have now retested with an updated setup:

- Libreelec 12.0 Nightly from February 16
- New __Beta3__ of Kodi 21.0

The outcome on my side is as follows:
- I tried from the folder popular shows. Playback is fine, but the subtitles do not appear. Still, Kodi shows that subtitles should be in swedish (SWE).
- When I try live broadcast channels, playback always stops after 32 seconds. Here, I noticed something new: once I interact with the system (e.g. I bring up the subtitle selection dialog, or audio selection dialog), playback continues until the next 32 seconds are reached.

**Comment by linqcan** [comment]
Hi @gandalth
Thanks for trying the beta version out on Kodi 21!
The issue you have seen with playback stalling after 32 seconds is an issue with `inputstream.adaptive`, the addon that manages the video playback in the upcoming version of this addon, on Kodi 21.

Most likely it is this bug you are encountering: https://github.com/xbmc/inputstream.adaptive/issues/1461

I'd recommend you to get off the Kodi 21 track (and use Kodi 20 stable with `inputstream.adaptive` 20.3.17) if you want to keep trying this addon's beta.

**Comment by gandalth** [comment]
Thank you, I am sure that's the bug I'm encountering. I will follow up!

I see that there is a proposed fix for inputadaptive. I will try to test with that as well.

**Comment by gandalth** [comment]
I am now testing on this system running Kodi 20:

RPi4
Libreelec 11.0.6 (01-29-2024)
Kodi 20.3.0 (Build 01-27-2024)

When installing the addon (with subfolder problem fixed), I get this error:

2024-02-25 08:31:55.916 T:954     error <general>: CAddonInstallJob[plugin.video
.svtplay]: The dependency on script.module.requests version 2.22.0 could not be 
satisfied.

Can I alter the sources myself to request a lower version and make it suitable for Kodi 20.3?

**Comment by linqcan** [comment]
Yes, you could do that if you have altered it to work for Kodi 21. Should
matter what version it is for the addon.

**Comment by gandalth** [comment]
Oops, my bad. It was an unresolved requirement on my system.

**Comment by gandalth** [comment]
Huzzah, subtitles are working.

I tested with live program of SVT 1 and SVT Barn.
Many thanks at @linqcan, this is such a great enabler to learn the Swedish language better. Nu kör vi, Sverige!

Quick summary for other users uninvolved so far:
- linqcan's release linked above works fine on Kodi 20.
- To install from zip, you need to repack his release into a folder.
- I will try to verify this release also with Kodi 21, once I can verify inputstream-adaptive's hotfix.

**Comment by gandalth** [comment]
I have now tested with Libreelec 12.0 Beta2 (11.95.2), which is
- Kodi 21.0 (Final)
- inputstream.adaptive 21.4.5

Thanks to the fixes in inputstream.adaptive, the playback bug is resolved. Using xbmc-svtplay-ia-beta-1.zip, I get subtitles in prerecorded shows, but not in live program (such as SVT 1 or SVT Barn). 

The subtitles in live programs work only in Kodi 20, as tested above (February 25).

**Comment by linqcan** [comment]
@gandalth Sorry for not being too responsive, there has been too much going on in life lately.

Thanks for doing the investigations you have done so far!

Based on the fact that the addon was not changed between your tries, I would assume there is some regression in Kodi 21.0 that you have hit (?).

Do you happen to have a [debug log](https://kodi.wiki/view/Log_file) from when the issue with live streams happens?

**Comment by gandalth** [comment]
No worries @linqcan .

Libreelec 12 is now final, let me retest and provide a debug log. It will take me two weeks though, this time.

**Comment by RosensRauk** [comment]
Coreelec 
Yatse
Android

Regarding surround audio. 
When I share a link from the svtplay-app (android) to coreelec 21 I can choose surround audio. From the GUI in Kodi I am always only offered stereo and no surround audio. Also, when using link sharing I always get subtitles, something that from direct play is iffy as best.
Thank you for creating this addon.
